extra/html-extra/HtmlExtension.php

```diff
@@ -108,4 +112,75 @@ function twig_html_classes(...$args): string
 
     return implode(' ', array_unique($classes));
 }
+
+function twig_attr_merge(...$arrays): array
+{
+    $result = [];
+
+    foreach ($arrays as $argNumber => $array) {
+        if (!$array) {
+            continue;
+        }
+
+        if (!twig_test_iterable($array)) {
+            throw new RuntimeError(sprintf('The attr_merge filter only works with arrays or "Traversable", got "%s" for argument %d.', \gettype($array), $argNumber + 1));
+        }
+
+        $array = twig_to_array($array);
+
+        foreach (['class', 'style', 'data'] as $deepMergeKey) {
+            if (isset($array[$deepMergeKey])) {
+                $value = $array[$deepMergeKey];
+                unset($array[$deepMergeKey]);
+
+                if (!twig_test_iterable($value)) {
+                    $value = (array) $value;
+                }
+
+                $value = twig_to_array($value);
+
+                $result[$deepMergeKey] = array_merge($result[$deepMergeKey] ?? [], $value);
+            }
+        }
+
+        $result = array_merge($result, $array);
+    }
+
+    return $result;
+}
+
+function twig_attr(Environment $env, ...$args): string
+{
+    $attr = twig_attr_merge(...$args);
+
+    if (isset($attr['class'])) {
+        $attr['class'] = trim(implode(' ', array_values($attr['class'])));
```

Comment:
The `array_values` is unnecessary because `implode` ignores the keys.

```suggestion
        $attr['class'] = trim(implode(' ', $attr['class']));
```